src/VisualStudio/Core/Def/RoslynPackage.cs

```diff
@@ -127,11 +131,16 @@ protected override async Task InitializeAsync(CancellationToken cancellationToke
             // Ensure the options persisters are loaded since we have to fetch options from the shell
             _componentModel.GetExtensions<IOptionPersister>();
 
-            _workspace = _componentModel.GetService<VisualStudioWorkspace>();
+            _workspace = (VisualStudioWorkspaceImpl)_componentModel.GetService<VisualStudioWorkspace>();
             _workspace.Services.GetService<IExperimentationService>();
 
             RoslynTelemetrySetup.Initialize(this, telemetrySession);
 
+            var provider = new VisualStudioDiagnosticAnalyzerProvider(this);
+
+            // fire and forget
+            _ = Task.Run(() => InitializeHostAnalyzerReferences(provider));
```

Comment:
If you need this to run on VisualStudioWorkspace creation, follow the pattern here:

https://github.com/dotnet/roslyn/blob/0a59b84dfe7c93f80c3f5b7fa791e8db244e2b95/src/VisualStudio/Core/Def/Implementation/LanguageClient/LanguageServerClientEventListener.cs#L22-L24

This stuff doesn't actually run until our package loads; in the case of CPS project loads that doesn't force our package to load. It's not until either a file is opened or Tools > Options is triggered where we'll get a package load triggered.

(And yes, some of this other stuff probably needs to move as well; it predates the existence of the new mechanism.)

Reply:
Done.